packages/react-spinbutton/src/components/SpinButton/SpinButton.types.ts

```diff
@@ -1,20 +1,163 @@
-import type { ComponentProps, ComponentState, IntrinsicShorthandProps } from '@fluentui/react-utilities';
+import type * as React from 'react';
+import type {
+  ComponentProps,
+  ComponentState,
+  IntrinsicShorthandProps,
+  ObjectShorthandProps,
+} from '@fluentui/react-utilities';
+import { LabelProps } from '@fluentui/react-label';
 
 export type SpinButtonSlots = {
   // TODO Add slots here and to spinButtonShorthandProps in useSpinButton.ts
+  /**
+   * The root element of SpinButton is a container `<div>`.
+   * The root slot receives the `className` and `style` specified on the `<SpinButton>`.
+   * All other native props are applied to the primary slot: `input`.
+   */
   root: IntrinsicShorthandProps<'div'>;
+
+  /**
+   * Input that displays the current value and accepts direct input from the user.
+   * Displayed value is formatted.
+   *
+   * This is the primary slot.
+   */
+  input: IntrinsicShorthandProps<'input'>;
+
+  /**
+   * Renders the increment button.
+   */
+  incrementButton: IntrinsicShorthandProps<'button'>;
+
+  /**
+   * Renders the decrement button.
+   */
+  decrementButton: IntrinsicShorthandProps<'button'>;
+
+  /**
+   * Renders the label.
+   */
+  label: ObjectShorthandProps<LabelProps> | IntrinsicShorthandProps<'label'>;
+};
+
+export type SpinButtonChangeData = {
+  /**
+   * New value after the change.
+   * E.g., `1`
+   */
+  value: number;
+
+  /**
+   * New formatted value after the change.
+   * E.g., `"$1.00"`
+   */
+  formattedValue: string;
+
+  /**
+   * Previous value, before the change.
+   */
+  prevValue: number;
+
+  /**
+   * Previous formatted value, before the change.
+   */
+  prevFormattedValue: string;
 };
 
+export type SpinButtonFormatter = (value: number) => string;
+export type SpinButtonParser = (formattedValue: string) => number;
+export type SpinButtonKeyboardDirection = 'none' | 'up' | 'down';
```

Comment:
Can you please clarify for what this type stays?

Reply:
I had planned to use it in the internal implementation of the SpinButton but I'm reconsidering how I want to approach that and, in any case, this probably shouldn't be exported.